File: Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        lc = strs[0]
        for i in range(1, len(strs)):
            j, k = 0, 0
            cw = strs[i]
            while j < len(cw) and k < len(lc) and cw[j] == lc[k]:
                j += 1
                k += 1
            lc = lc[:k]
        return lc
                

File: Longest-Common-Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let lc = strs[0]

    for (let i=1; i < strs.length; i++){
        let cw = strs[i]
        let j = 0
        let k = 0
        while ( j < cw.length && k < lc.length && cw[j] == lc[k]){
            j += 1
            k += 1
        }
        lc = lc.substring(0,k)
    }
    return lc
};